libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookEntry.cs

```diff
@@ -23,18 +23,21 @@ public class FacebookEntry
         /// Gets or sets the messaging list.
         /// </summary>
         /// <value>List containing one messaging object. Note that even though this is an enumerable, it will only contain one object.</value>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "it needs to be set in ActivityToFacebook method")]
```

Comment:
I think that you can still declare this as 
```
public List<FacebookMessage> Messaging { get; } = new List<FacebookMessage>();
```
and remove the rule exclude, the JsonConvert should be able to use the existing instance of the list.